### PR TITLE
allow `RepaintSignal: ?Sync`

### LINKED ISSUE #1186: `RepaintSignal: Sync`
`epi::backend::RepaintSignal` requires  `Sync` (introduced in #82), which requires wrapping `!Sync` types (`winnit::EventLoopProxy` for example) with a mutex. Not cool IMO.

### PR DESCRIPTION
**Pull request by ZetaNumbers**
Closes #1186.

### CODE REVIEW

File: egui_web/src/backend.rs [hunk 1]
```diff
@@ -55,11 +55,12 @@ impl WebInput {
 
 use std::sync::atomic::Ordering::SeqCst;
 
-pub struct NeedRepaint(std::sync::atomic::AtomicBool);
+#[derive(Clone)]
+pub struct NeedRepaint(std::sync::Arc<std::sync::atomic::AtomicBool>);
```
Comment (Mingun) [inline egui_web/src/backend.rs:59]: Atomic reference counter for an atomic variable seems an overkill.

Comment (ZetaNumbers) [reply egui_web/src/backend.rs:59]: It has been there already, i just moved it. ¯\\\_(ツ)\_/¯

**Comment by emilk** [comment]
Is all this new code just to avoid the performance impact of locking a mutex when the users requests a repaint? Or is there some other benefit I'm not seeing?

**Comment by ZetaNumbers** [comment]
> Is all this new code just to avoid the performance impact of locking a mutex when the users requests a repaint? Or is there some other benefit I'm not seeing?

This PR firstly fixes logic of the library and only then the performance. There are several of benefits to it:
- Avoids mutex poisoning and panic deadlocks (i. e. if some thread panics while using `request_repaint`, then other threads would panic on a `request_repaint` too);
- Use of DST eliminates an indirection;
- Avoids unnecessary locks (majority of channel `Sender`s (like mpsc) are `Send + !Sync` which are used by window handling libraries);
- There may be other downsides of relying on mutex, like if it's not implemented for some platform or another indirection (posix targets store mutex on the heap IIRC).

**Pull request closed without merging**